BookingsApi/BookingsApi.Domain/JusticeUser.cs

```diff
@@ -81,5 +82,30 @@ public bool IsDateBetweenWorkingHours(DateTime startDate, DateTime endDate, Allo
             return (workHourStartTime <= startDate.TimeOfDay || configuration.AllowHearingToStartBeforeWorkStartTime) && 
                    (workHourEndTime >= endDate.TimeOfDay || configuration.AllowHearingToEndAfterWorkEndTime);
         }
+
+        public void Delete()
```

Comment:
This method name should describe what it's doing. I think something like 
```suggestion
        public void UnallocateHearingsAndDeleteHours()
```
is more verbose and wouldn't stop me from having to jump into the code to see what Delete does

Reply:
I called it `Delete` as that is our name for the operation on the user (as per our other deletions). The deallocation is an implementation detail of that operation, so not sure about including it in the name, but happy to discuss further